pyDrop/clustering/PCRData.py

```diff
@@ -0,0 +1,444 @@
+    >>> data = PCRData(filepath, sep=",", header=0)
+    >>> data.plot()
+    PCRData also contains functionality to subsample data if true labels are given for testing purposes:
+    >>> data.subsample_clusters({1: 1.0, 2: 0, 3: 0.5})
+    """
+    def __init__(self, data, X_columns: list=[], y_column: str="", verbose: bool=False, *args, **kwargs):
+        if isinstance(data, pd.DataFrame):
+            pass
+        elif isinstance(data, str):
+            try:
+                data = pd.read_csv(data, *args, **kwargs)
+            except Exception as e:
+                raise PCRDataReadingException(e)
+        else:
+            raise PCRDataReadingException(f"Data must be of the format str or pandas.DataFrame. Type {type(data)} was given.")
+        
+        self.verbose = verbose
+        
+        if X_columns: # explicitly given X columns
+            if any([name not in data.columns for name in X_columns]): # if not all of X_columns appear 
+                ghost_names = [name for name in X_columns if name not in data.columns]
+                raise FeaturesImproperlySpecified(f"The given names do not appear in the data feature columns: {ghost_names}")
+            else:
+                self.X_cols = X_columns
+        else: # if no explicitly given X columns, try to infer based on a standard format ch*
+            regexp = re.compile(r'ch*', re.I)
+            skipped_cols = []
+            for name in data.columns:
+                match = regexp.search(name)
+                if match and match.span()==(0,2):
+                    X_columns.append(name)
+                else:
+                    skipped_cols.append(name)
+            if skipped_cols and self.verbose:
+                raise UserWarning(f"The following columns were skipped when assigning features automatically {skipped_cols}")
+            
+            self.X_cols = X_columns
+
+        self.supervised = True
+        if y_column:
+            if y_column not in data.columns:
+                raise LabelsImproperlySpecified(f"The given y column ({y_column}) does not appear in the data.")
+            else:
+                self.y_col = y_column
+        else: # if no explicitly given y column, try to infer based on the format column_*
+            regexp = re.compile(r'cluster*', re.I)
+            self.y_col = ""
+            for name in data.columns:
+                if regexp.search(name) and not self.y_col:
+                    self.y_col = name
+                elif regexp.search(name) and self.y_col:
+                    raise LabelsImproperlySpecified(f"2 y columns found: [{self.y_col}, {name}]. Please use the y_column argument to specify which column to use. ")
+            if not self.y_col:
+                raise UserWarning("No y column found. Now in unsupervised mode.")
+                self.supervised = False # assign false only if no default column found
+
+        self.col_to_id = dict([(col_name, idx) for idx, col_name in enumerate(self.X_cols)])
```

Comment:
```suggestion
        self.col_to_id = {col_name: idx for idx, col_name in enumerate(self.X_cols)}
```
If you're not trying to be 2.* compatible, you can just use a dictionary comprehension.